lib/rest/routes/pipelines.js

```diff
@@ -0,0 +1,30 @@
+const express = require('express');
+
+module.exports = function ({config}) {
+  let router = express.Router();
+  router.put('/:name', function (req, res, next) {
+    let isUpdate;
+    config.update(json => {
+      isUpdate = req.params.name in json.pipelines;
+      json.pipelines[req.params.name] = req.body;
+      return json;
+    })
+    .then(() => res.status(isUpdate ? 204 : 201).send())
+    .catch(next);
+  });
+  router.get('/:name', function (req, res, next) {
+    res.json(config.gatewayConfig.pipelines[req.params.name]);
+  });
+  router.get('/', function (req, res, next) {
```

Comment:
Is the `next` parameter needed for all these routes? This can probably be safely removed.

Reply:
and what is the problem with having next ? 
(req,res), or (req,res,next) are well known signatures for middleware 

and it seems that there is no rule in standard js that says it is not good
https://standardjs.com/rules.html

I remember having that rule enabled in the project but I didn't like  the experience, because in some cases implementation change and you need add\remove arguments constantly. 

but I will agree if this will promoted as ESlint rule

Reply:
The general rule I found among the project (although it's not codified) is

route: `fn(req, res)`
middleware: `fn(req, res, next)`

but it's a nitpick, it's absolutely not blocking anything.